File: LeetCode/Easy/Count the Number of Consistent Strings/1684.js
/**
 * @param {string} allowed
 * @param {string[]} words
 * @return {number}
 */
var countConsistentStrings = function(allowed, words) {
    const set = new Set(allowed);
    let count = words.length;

    for (const word of words) {
        for (const w of word) {
            if (!set.has(w)) {
                count--;
                break;
            }
        }
    }

    return count;
};

File: LeetCode/Easy/Count the Number of Consistent Strings/1684.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        boolean[] isAllowed = new boolean[26];
        for (char a : allowed.toCharArray()) isAllowed[a - 'a'] = true;

        int count = words.length;
        for (String word : words) {
            for (char w : word.toCharArray()) {
                if (!isAllowed[w - 'a']) {
                    count--;
                    break;
                }
            }
        }
        
        return count;
    }
}
